subprojects/core/src/main/java/org/gradle/api/internal/tasks/TaskValidationContext.java

```diff
@@ -19,6 +19,10 @@
 import org.gradle.api.internal.file.FileResolver;
 import org.gradle.internal.reflect.ParameterValidationContext;
 
+import java.io.File;
+
 public interface TaskValidationContext extends ParameterValidationContext {
     FileResolver getResolver();
+
+    boolean isInReservedFileLocation(File file);
```

Comment:
Nit: `File` is a bit redundant in the name.

Reply:
Parameter should be called `directory` and maybe we should call the method `isInReservedDirectory()`.

Reply:
The parameter can be a file or a directory. So I guess we should call the parameter `location` or `path`?

Reply:
The method is now called `isInReservedDirectory` and the parameter `location`.

Reply:
👍